File: vdp-java/src/test/java/com/visa/vdp/vctc/TestValidation.java
package com.visa.vdp.vctc;

import java.util.HashMap;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.Test;

import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;

public class TestValidation {

    VisaAPIClient visaAPIClient;

    @Test(groups = "vctc")
    public void setUp() {
        this.visaAPIClient = new VisaAPIClient();
    }

    @Test(groups = "vctc")
    public void testRetreiveListofDecisionRecords() throws Exception {
        String baseUri = "vctc/";
        String resourcePath = "validation/v1/decisions/history";
        String queryParams = "?limit=1&page=1";

        CloseableHttpResponse response = this.visaAPIClient.doMutualAuthRequest(baseUri + resourcePath + queryParams, "Retrieve List of Recent Decision Records", "", MethodTypes.GET, new HashMap<String, String>());
        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
        response.close();
    }
}

File: visa-dev-samplecode-master/vdp-python/visa/test/vctc/test_validation.py
from visa.helpers.visa_api_client import VisaAPIClient
import unittest
'''
@author: visa
'''

class TestValidationAPI(unittest.TestCase):
    
    def setUp(self):
        self.visa_api_client = VisaAPIClient()

    def test_retrieve_list_recent_decision_records(self):
        base_uri = 'vctc/';
        resource_path = 'validation/v1/decisions/history';
        query_string = '?limit=1&page=1'
        response = self.visa_api_client.do_mutual_auth_request(base_uri + resource_path + query_string, '', 'Retrieve List of Recent Decision Records call', 'get')
        self.assertEqual(str(response.status_code) ,"200" ,"Retrieve List of Recent Decision Records test failed")
        pass
